test/OrchardCore.Tests/Apis/Lucene/LuceneQueryTests.cs

```diff
@@ -152,5 +155,127 @@ public async Task TwoWildcardQueriesWithBoostHasResults()
                 Assert.Contains("Orchard", contentItems.ElementAt(3).DisplayText, StringComparison.OrdinalIgnoreCase);
             };
         }
+
+
+
+
+        [Fact]
+        public async Task ShouldUpdateIndexContentAfterContentMangerAction()
+        {
+            using var context = new BlogContext();
+            await context.InitializeAsync();
+
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+                var contentManager = scope.ServiceProvider.GetRequiredService<IContentManager>();
+
+                var existContent = await contentManager.GetAsync(context.BlogContentItemId);
+
+                var firstContent = await contentManager.CloneAsync(existContent);
+                firstContent.Alter<AutoroutePart>(part =>
+                {
+                    part.Path = "test1";
+                });
+                var luceneQuery = new LuceneQuery
+                {
+                    Index = "Search",
+                    ReturnContentItems = false,
+                    Template = @"
+                            {
+                                ""query"": {
+                                    ""term"": { ""Content.ContentItem.ContentType"": ""BlogPost"" }
+                                },
+                                ""filter"": [
+                                    {
+                                        ""term"": { ""Content.ContentItem.ContentItemId"":""{{parameters.contentItemId}}"" }
+                                    }
+                                ]
+                            }",
+                };
+                var queryParameters = new Dictionary<string, object>
+                {
+                    ["contentItemId"] = firstContent.ContentItemId
+                };
+
+                #region Create a content item
+
+                
+                //await contentManager.CreateAsync(firstContent);
+                //var contentFromDb = await contentManager.GetAsync(firstContent.ContentItemId);
+                //Assert.NotNull(contentFromDb); faild
```

Comment:
@Skrypt 
Our content item operation is a bit complicated~~

Reply:
Hi @Piedone ，need your help , I don't know what to do..
https://github.com/OrchardCMS/OrchardCore/issues/15601#issuecomment-2090289841

Reply:
What's your question, specifically?

Reply:
Just as @Skrypt  mentioned in this issue at https://github.com/OrchardCMS/OrchardCore/issues/15601 ,
I understand that we should verify the impact on indexing functionality by separately testing Create, Update, Draft, and Publish actions with ContentManager. 

However, based on the current design of `DefaultContentManager` which requires calling a sequence of methods to save data to the database and trigger Lucene indexing, it seems difficult if not impossible to isolate tests for just the Create and Update actions.

All that aside, am I currently unit testing the right way?

Reply:
You don't need to test `DefaultContentManager` or have a full indexing test. Rather, unit tests on `LuceneIndexingContentHandler` with mocks is sufficient. I.e, the tests need to check that `LuceneIndexingContentHandler` behaves correctly based on what it calls on the mocks.

If you search for "mock.setup" and "mock.verify" in the codebase, you'll see how this works. A lot of tests do something similar.